File: 954-maximum-sum-circular-subarray/maximum-sum-circular-subarray.cs
public class Solution {
    public int MaxSubarraySumCircular(int[] nums) {
        int totalSum = 0;
        int maxSum = int.MinValue;
        int currentMax = 0;
        int minSum = int.MaxValue;
        int currentMin = 0;

        foreach (int num in nums) {
            totalSum += num;

            currentMax = Math.Max(currentMax + num, num);
            maxSum = Math.Max(maxSum, currentMax);

            currentMin = Math.Min(currentMin + num, num);
            minSum = Math.Min(minSum, currentMin);
        }

        if (maxSum < 0) {
            return maxSum;
        }

        return Math.Max(maxSum, totalSum - minSum);
    }
}

File: 954-maximum-sum-circular-subarray/maximum-sum-circular-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubarraySumCircular = function(nums) {
    let maxSum=nums[0],minSum=nums[0], total=nums[0], curMax=nums[0],curMin=nums[0];
    for (let i=1; i<nums.length; i++){
        curMax=Math.max(nums[i], curMax+nums[i]);
        maxSum=Math.max(maxSum, curMax);
        curMin=Math.min(nums[i],curMin+nums[i]);
        minSum=Math.min(minSum,curMin);
        total +=nums[i];
    }
    return maxSum > 0 ? Math.max(maxSum, total-minSum):maxSum;
};